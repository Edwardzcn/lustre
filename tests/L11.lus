node nats() returns (n : int)
let
  n = 0 -> 1 + pre n;
tel

node f(c : bool) returns (y : int)
let
  y = condact(c, nats(), 7);
tel

