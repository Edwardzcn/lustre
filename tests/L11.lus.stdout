PhaseToCore
===========

node (c : bool)
  returns (n)
  assumes ()
  shows ()
let
  pre_nats : int when c
    = pre nats
   
  nats : int when c
    = 0 -> Add(1, pre_nats)
   
  current_nats : int
    = current nats
   
  pre_n : int
    = pre n
   
  init_pre_n : int
    = 42 -> pre_n
   
  n : int
    = ITE(c, current_nats, init_pre_n)
tel

--- Step 1 ---
c : bool  = False
n = 42
--- Step 2 ---
c : bool  = False
n = 42
--- Step 3 ---
c : bool  = True
n = 0
--- Step 4 ---
c : bool  = False
n = 0
--- Step 5 ---
c : bool  = False
n = 0
--- Step 6 ---
c : bool  = True
n = 1
--- Step 7 ---
c : bool  = True
n = 2
--- Step 8 ---
c : bool  = False
n = 2
--- Step 9 ---
c : bool  = False
n = 2
--- Step 10 ---
c : bool  = False
n = 2
--- Step 11 ---
c : bool  = (EOF)
